storage/badger/operation/common.go

```diff
@@ -65,7 +65,7 @@ func insert(key []byte, entity interface{}) func(*badger.Txn) error {
 			return storage.ErrAlreadyExists
 		}
 		if err != badger.ErrKeyNotFound {
-			return fmt.Errorf("could not check key: %w", err)
+			return storage.ErrNotFound
```

Comment:
⚠️ **I think the previous implementation was correct**

In my option, this is one of the few cases, where we should **not change the error type**. The function `insert` proceeds as follows:
1. try to read the key and error if this succeeds, because we don't want to override an existing value: https://github.com/onflow/flow-go/blob/35d66d677458e61f01df75eed282eef6d3b80732/storage/badger/operation/common.go#L62-L66 so The only way we reach the subsequent code is if `tx.Get(key)` returned an error. For the happy path, we _expect_ an `badger.ErrKeyNotFound`. 
2.  The code here handles the _unhappy_  path, i.e. we received an error _other_ than the expected `badger.ErrKeyNotFound`. In this case, the method should fail, but _not_ with `storage.ErrNotFound` because this is not what happens. 

The case where we encounter a `badger.ErrKeyNotFound` is the happy path for which we insert the `entity`.

Reply:
Good catch!